stingray/utils.py

```diff
@@ -975,3 +986,134 @@ def find_nearest(array, value):
+                                   overwrite=True)
+
+    meta_data_group.create_dataset(name='err_dist',
+                                   data=lc.err_dist,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    meta_data_group.create_dataset(name='mjdref',
+                                   data=lc.mjdref,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+
+def _saveChunkEV(ev, fname):
+    """
+    Prepare EventList for temporary saving.
+
+    Parameters
+    ----------
+    ev: :class:`stingray.events.EventList` object
+        EventList to be saved
+
+    fname: str
+        High Level diretory name where lightcurve is to be saved.
+    """
+    # Creating a Nested Store and multiple groups for temporary saving
+    store = zarr.NestedDirectoryStore(fname)
+    ev_data_group = zarr.group(store=store, overwrite=True)
+    main_data_group = ev_data_group.create_group('main_data', overwrite=True)
+    meta_data_group = ev_data_group.create_group('meta_data', overwrite=True)
+
+    compressor = Blosc(cname='lz4', clevel=1, shuffle=-1)  # Tested
+
+    # REVIEW: Max chunk size can be 8388608 or 2**23. This efficiently balances time, memory. Memory consumption restricted to 9.1 GB
+    main_data_group.create_dataset(name='times',
+                                   data=ev.time,
+                                   compressor=compressor,
+                                   overwrite=True,
+                                   chunks=(8388608, ))
+
+    meta_data_group.create_dataset(name='dt',
+                                   data=ev.dt,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    meta_data_group.create_dataset(name='ncounts',
+                                   data=ev.ncounts,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    meta_data_group.create_dataset(name='mjdref',
+                                   data=ev.mjdref,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    if ev.energy:
+        main_data_group.create_dataset(name='energy',
+                                       data=ev.energy,
+                                       compressor=compressor,
+                                       overwrite=True,
+                                       chunks=(8388608, ))
+
+    # FIXME: GTI's are not consistently saved
+    if ev.gti:
+        main_data_group.create_dataset(name='gti', data=ev.gti, overwrite=True)
+
+    if ev.pi:
+        main_data_group.create_dataset(name='pi_channel',
+                                       data=ev.pi,
+                                       compressor=compressor,
+                                       overwrite=True)
+
+
+def saveData(data_obj, f_name):
```

Comment:
In my opinion, what we would need to do for that is to read a part of the raw data, convert it into a lightcurve and then save it on the disk. Perform the action repeatedly until all the raw data is saved as small lightcurves(essentially chunks) on the disk.

Methods to achieve the same:
1. Use lazy loading of the data using generators or something, reading the raw data stored on the disk and converting it into Lightcurves and then saving these lightcurves on the disk as I mentioned above.
2. Also, can this be done using dask, by lazy evaluation(I am half sure that it can work)? But for that, we would need to change the whole Lightcurve code from NumPy to Dask.